src/EFCore.PG/Query/Sql/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -41,6 +42,15 @@ public class NpgsqlQuerySqlGenerator : DefaultQuerySqlGenerator
     {
         readonly bool _reverseNullOrderingEnabled;
 
+        /// <summary>
+        /// The collection of standard comparison operators.
+        /// </summary>
+        /// <remarks>
+        /// See: https://github.com/aspnet/EntityFrameworkCore/issues/9143
+        /// </remarks>
+        [NotNull] static readonly HashSet<string> StandardComparisonSymbols =
```

Comment:
Perf isn't a huge factor here (because generated SQL is cached by EF Core), however:
In theory of course a contains operation is more efficient on a `HashSet`. In practice, however, for small sets it's much more efficient to simply use an array, because scanning through a local array and doing equality is faster than the overhead required by `HashSet` (calculating the hashcode, traversing the internal data structure...). You can do a quick BenchmarkDotNet test to see.